src/pybind/mgr/dashboard/services/orchestrator.py

```diff
@@ -64,8 +64,8 @@ def get(self, hostname: str) -> Optional[HostSpec]:
         return hosts[0] if hosts else None
 
     @wait_api_result
-    def add(self, hostname: str):
-        return self.api.add_host(HostSpec(hostname))
+    def add(self, hostname: str, status: str):
+        return self.api.add_host(HostSpec(hostname, status=status))
```

Comment:
sorry... i do not have a good knowledge of the dashboard coed.. but if this is calling the ceph orchestrator... i think this is not the right signature:
take a look to:
https://github.com/ceph/ceph/blob/fdb2aadc5078c8c3aaddad4e61cf9ba93bf770b4/src/pybind/mgr/orchestrator/module.py#L326

Reply:
I think its adding the host specs to the HostSpec file. IMHO the current implementation is good and easy.

Reply:
@epuertat What do you think?

Reply:
I would suggest to change it to this 
`def add(self, hostname: str, status: Optional[str] = None):
        return self.api.add_host(HostSpec(hostname, status))`

Reply:
I prefer the idea of consuming an object-oriented API (`HostSpec`) vs. a functional API (`add_host(hostname, addr, labels, maintenance)`. Is that the debate here?

The OOP approach enforces encapsulation and allows dashboard to perform introspection on foreign classes (like the orchestrator ones) rather than creating proxies and duplicate wrappers all around.